lib/emque/producing/publisher/google_cloud_pubsub.rb

```diff
@@ -0,0 +1,47 @@
+require "google/cloud/pubsub"
+
+module Emque
+  module Producing
+    module Publisher
+      class GoogleCloudPubsub < Emque::Producing::Publisher::Base
+        def initialize(project_id:, credentials:)
+          self.pubsub = Google::Cloud::PubSub.new(
+            :project_id => project_id,
+            :credentials => credentials
+          )
+        end
+
+        def publish(topic_name, message_type, message, raise_on_failure)
+          Emque::Producing.logger.debug("GoogleCloudPubsub#publish")
+
+          topic = pubsub.topic(message_type)
```

Comment:
need a more granular topic name in google pubsub to split up messages

Reply:
@Peterortiz82 should we expand on this naming pattern? since there are only one level of topics in gcp pubsub, perhaps something more like `"emque.#{message_type}"` to better group them in the topic list and make it more clear what's publishing to the topics?

Reply:
I think that's a good idea. I could imagine a future where messages are being published in different ways and it will be nice to quickly identify which ones are published from emque.